### ISSUE TITLE
The default_action of network_rules in azurerm_storage_account isn't working as expected

### ISSUE THREAD

**Issue by hugo-paredes** [hashicorp/terraform-provider-azurerm#17195]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.2

### AzureRM Provider Version

3.9.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "storage_account" {
  name                            = "somestorage"
  location                        = "West Europe"
  resource_group_name             = "existing-resource-group"
  account_replication_type        = "RAGRS"
  account_tier                    = "Standard"
  account_kind                    = "StorageV2"
  enable_https_traffic_only       = true
  allow_nested_items_to_be_public = false
  min_tls_version                 = "TLS1_2"

  blob_properties {
    versioning_enabled       = false
    change_feed_enabled      = false
    last_access_time_enabled = false
  }

  network_rules {
    default_action = "Deny"
    virtual_network_subnet_ids = ["valid-subnet-id"]
    ip_rules = ["IP_RANGE"]
  }

  tags = local.tags
}
```

### Debug Output/Panic Output

```shell
│ Error: retrieving Container "ex1" (Account "somestorage" / Resource Group "existing-resource-group"): containers.Client#GetProperties: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailure" Message="This request is not authorized to perform this operation.\nRequestId:1518f7d3-b01e-0032-37a3-7c5bdb000000\nTime:2022-06-10T08:24:10.1652623Z"
│
│   with module.storage_account.azurerm_storage_container.container["ex1"],
│   on ..\..\main.tf line 39, in resource "azurerm_storage_container" "container":
│   39: resource "azurerm_storage_container" "container" {
```

### Expected Behaviour

I'm running the code from a laptop connected to a work VPN. I've whitelisted the IP range in the network rules section. My understanding from the documentation is that if you set the `default_action` to `Deny`, it would deny access from anything that's not configured in the network rules section.
If I whitelist the IP range and a subnet that has access to it (for internal things), I should be able to access it and make changes to the resource.

### Actual Behaviour

I get an un-authorized access when running `terraform plan`.

### Steps to Reproduce

1 .`terraform init`
2. `terraform plan -out .\.terraform\terraform.tfplan`
3. terraform apply .\.terraform\terraform.tfplan

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#network_rules

**Comment by tombuildsstuff** [comment]
Duplicate of #2977 - in short the Storage Account Resource has to reach out to the Data Plane API here, #2977 is tracking conditionally using the Resource Manager API instead (which'll fix this) - but the two API's aren't currently feature-compatible and this is currently blocked on the Azure Storage Team. Would you mind subscribing to #2977 for updates?

Thanks!
